cli/tests/unit/cache_api_test.ts

```diff
@@ -173,3 +173,20 @@ Deno.test(async function cachePutFailedBody() {
   // if it fails to read the body, the cache should be empty
   assertEquals(response, undefined);
 });
+
+Deno.test(async function cachePutOverwrite() {
+  const cacheName = "cache-v1";
+  const cache = await caches.open(cacheName);
+
+  const request = new Request("https://example.com/overwrite");
+  const res1 = new Response("res1");
+  const res2 = new Response("res2");
+
+  await cache.put(request, res1);
+  const res = await cache.match(request);
+  assertEquals(await res?.text(), "res1");
+
+  await cache.put(request, res2);
+  const res_ = await cache.match(request);
+  assertEquals(await res_?.text(), "res2");
+});
```

Comment:
The test passes on main

Reply:
Run it twice and it will fail.

Reply:
It passes the second run too

Reply:
Unfortunately, we need to preserve sqlite metadata to reproduce the bug and it doesn't work for unit tests as every invocation gets new sqlite db file. I created integration tests that fail on main: https://github.com/denoland/deno/pull/18649/commits/515d599b4a71edaba793cf7f40959fd5a2093f5e